File: domo/relay.script.d/0.php
<?php

/*
echo $r['id']; // affiche l'id du relay (= au nom du fichier $ID.php)

echo $r['etat']; // affiche l'état du relay $ID 
    // Etat : 
    //  - 1 : off auto
    //  - 2 : on auto

echo $d['SOC']; // affiche l'état de charge (en %) du parc batterie - lu sur me BMV
echo $d['P']; // affiche la puissance instantané (valeur possiblement négative) - lu sur me BMV

// A la fin du script on retour 1 ou 2 pour redéfinir l'état du relais

return 1; // mettra le relais à off si ce n'est pas déjà le cas
*/
/*
$pingHost=array("192.168.1.10","192.168.1.12");

function checkIfComputerIsUp($pingHost) {
    $onlineHost=0;
    if (is_file('/usr/bin/fping')) {
        $hosts=null;
        foreach ($pingHost as $host) {
            $hosts.=' '.$host;
        }
        @exec("fping -c1 -t500 " . $hosts, $output, $result);
        if (isset($output[0])) {
            $onlineHost++;
        }
    } else {
        foreach ($pingHost as $host) {
            @exec("ping -c 1 -i 0.2 -W 1 " . $host, $output, $result);
            if ($result == 0) {
                $onlineHost++;
            }
        }
    }
    return $onlineHost;
}



// Si c'est éteind, faut-il l'allumer ?
if ($r['etat'] == 1) {
    // Si les batteries sont pleinnes
    if ($d['SOC'] == 100 ||
    // OU Si il est plus de 11 heure et que les batterie sont suppérieur à 95 c'est qu'il fait beau...
    date('G') > 11 && $d['SOC'] > 95) {
        // On enregistre la date du dernier allumage
        if (empty($timeLastUp[$r['id']])) {
            $timeLastUp[$r['id']] = time();
        }
        // On allume !
        return 2;
    } else {
        return 1;
    }
// Si c'est allumé, faut-il l'éteindre ?
} else if ($r['etat'] == 2) {
    $checkIfComputerIsUp = checkIfComputerIsUp($pingHost);
    // S'il est plus de 17h et qu'il n'y a pas d'ordinateur d'allumé
    if ((date('G') >= 17 && $checkIfComputerIsUp == 0) 
    // Si les batteries pass sous les 95% et qu'il n'y a pas d'ordinateur d'allumé
    || ($d['SOC'] <= 95 && $checkIfComputerIsUp == 0)) {

        // On enregistre la date de la dernière extinction
        if (empty($timeLastDown[$r['id']])) {
            $timeLastDown[$r['id']] = time();
        }
        // On éteind !
        return 1;
    } else {
        return 2;
    }
    
} 

*/
return 2;
?>

File: domo/relay.script.d/0.py

####################################
# Ce script est un exemple pour vous
# il est a adapté a vos besoin
####################################

# Script pour ma box internet

# config
pingHost=["192.168.1.10","192.168.1.12"]

def checkIfComputerIsUp(pingHost) :
    onlineHost=0
    pingBin="ping -i 0.2 -W 1 -c1 "
    if os.path.isfile('/usr/bin/fping'): 
        pingBin='/usr/bin/fping -c1 -t500 '
    logMsg(5, "Ping bin utilisé : " + pingBin)
    for host in pingHost:
        logMsg(5, pingBin + host + '>/dev/null 2>/dev/null')
        response = os.system(pingBin + host + '>/dev/null 2>/dev/null')
        #and then check the response...
        if response == 0:
            onlineHost=onlineHost+1
    logMsg(5, 'onlineHost ' + str(onlineHost))
    return onlineHost

timeUp=600

# Si il est éteind, faut-il l'allumer ?
if relayEtat[relayId] == 1:
    # Si le régulateur dit que c'est bientôt la fin de charge et qu'il est plus de 11h c'est qu'il va faire beau !
    if MpptAbsOrFlo(xmlData['CS']):
        returnLog='UP Le régulateur est en mode abs ou float'
        returnEtat=2
    if float(xmlData['SOC']) > 93 and int(time.strftime ('%H')) > 11 and int(time.strftime ('%H')) < 17:
        returnLog='UP La batterie est chargé à plus de 93% et qu\'il est entre 11 et 17h'
        returnEtat=2
# Si il est allumé, faut-il l'éteindre ?
elif relayEtat[relayId] == 2:
    nbComputerUp=checkIfComputerIsUp(pingHost)
    # S'il n'y a plus d'ordinateur d'allumé et que les batterie sont sous les 95% ou qu'il est après 17h on éteind
    if (nbComputerUp == 0 and float(xmlData['SOC']) <= 93):
        returnLog='DOWN pas d`ordinateur connecté et les batterie sous 93%'
        returnEtat=1
    if (nbComputerUp == 0 and int(time.strftime ('%H')) >= 17) :
        returnLog='DOWN pas d`ordinateur connecté et il est plus de 17h'
        returnEtat=1
    if timeUpMin(timeUp):
        returnLog='On maintient allumé, '
        returnEtat=2
